src/aql_graph_loader.rs

```diff
@@ -0,0 +1,1068 @@
+use crate::DatabaseConfiguration;
+use crate::client::auth::handle_auth;
+use crate::client::config::ClientConfig;
+use crate::client::{build_client, make_url};
+use crate::errors::GraphLoaderError;
+use bytes::Bytes;
+use reqwest::StatusCode;
+use reqwest_middleware::ClientWithMiddleware;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use tokio::task::JoinSet;
+
+/// Data types supported for graph attributes
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum DataType {
+    Bool,
+    String,
+    U64,
+    I64,
+    F64,
+    JSON,
+}
+
+/// Description of a data item (attribute) with name and type
+#[derive(Clone, Debug)]
+pub struct DataItem {
+    pub name: String,
+    pub data_type: DataType,
+}
+
+/// A batch of graph data containing vertices and edges
+#[derive(Debug)]
+pub struct GraphBatch {
+    /// Vertex IDs as byte vectors
+    pub vertex_ids: Vec<Vec<u8>>,
+    /// Vertex attributes, parallel to vertex_ids
+    pub vertex_attributes: Vec<Vec<Value>>,
```

Comment:
Would a name like `vertex_values` be more suitable here? When I read attributes I immediately think about the `key` rather than the `value`.

Reply:
I have renamed the members to `vertex_attribute_values` and `edge_attribute_values` resp. in the callback struct. In the latter example, where it is `Vec<DataItem>`, I think the name `vertex_attributes` is more appropriate, so I left this.